dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/instrumentation/MainLooperLongTaskStrategy.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.instrumentation
+
+import android.content.Context
+import android.os.Looper
+import android.util.Printer
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.internal.monitor.AdvancedRumMonitor
+import com.datadog.android.rum.tracking.TrackingStrategy
+import java.util.concurrent.TimeUnit
+
+internal class MainLooperLongTaskStrategy(
+    internal val thresholdMs: Long
+) : Printer, TrackingStrategy {
+
+    private val thresholdNS = TimeUnit.MILLISECONDS.toNanos(thresholdMs)
+    private var startUptimeNs: Long = 0L
+    private var target: String = ""
+
+    // region TrackingStrategy
+
+    override fun register(context: Context) {
+        Looper.getMainLooper().setMessageLogging(this)
+    }
+
+    override fun unregister(context: Context?) {
+        Looper.getMainLooper().setMessageLogging(null)
+    }
+
+    // endregion
+
+    // region Printer
+
+    override fun println(x: String?) {
+        if (x != null) {
+            detectLongTask(x)
+        }
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun detectLongTask(message: String) {
+        val now = System.nanoTime()
+        if (message.startsWith(PREFIX_START)) {
+            target = message.substring(PREFIX_START_LENGTH)
```

Comment:
the way the start message looks like is: `">>>>> Dispatching to " + msg.target + " " +msg.callback + ": " + msg.what`. I guess we could remove the last part after `:` and keep only the target, what do you think ?

Reply:
No, because the `what` actually has a meaning for people using `Handler` and `Message`. Removing it means losing information.